dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,203 @@
+# Minimal OS for Fobnail project
+
+The Fobnail project aims to provide a reference architecture for building
+offline integrity measurement servers on the USB device and clients running in
+Dynamically Launched Measured Environments (DLME). It should allow the Fobnail
+owner to verify the trustworthiness of the running system before performing any
+sensitive operation.
+
+According to the information provided at TrenchBoot Summit 2021 (the fragment
+discussed here - https://youtu.be/xZoCtNV8Qs0?t=5062), the minimum OS, together
+with the connected Fobnail Token, should allow determining the state of the
+platform before the target OS is launched. As we can see in the presentation,
```

Comment:
@TomaszAIR this is early launch example, in future we plan to do late launch implementation, what IMO is important to mention. Both early launch and late launch use cases were discussed many times on various occasions:
- https://youtu.be/pZF-jyJWTE4

Also I see no reason why Fobnail couldn't attest SRTM: https://www.youtube.com/watch?v=Eip5Rts6S2I

Reply:
@TomaszAIR I would like you to:
* [ ] distinguish between early launch and late launch - maybe TrenchBoot documentation or friends from EFT would be helpful
* [ ] mention that current approach is about early launch and late launch is for future improvements
* [ ] mention that not only DRTM PCRs can be attested but also SRTM PCRs, but for the purpose of DLME OS selection we care only about DRTM PCRs attestation by Fobnail, why it is important? Because without DLME OS we can have some weak claims  about platform state based on SRTM PCRs, this should be clear for the reader

Reply:
Thank you, that's helpful. We will address that.

Reply:
@pietrushnic I addressed two first points.
Regarding the third point, we are actually attesting both SRTM and DRTM, however that could be easily changed.

Reply:
@arturkow2000 policy should determine that.

Reply:
@pietrushnic It is already determined by policy, but policy is hardcoded currently.

Reply:
@arturkow2000 and that is fine, we just need internal plan for customizable policies. Do we have one?

Reply:
@pietrushnic We do plan to implement it, but we don't have any plan how it should be implemented.

Reply:
Related issue: https://github.com/fobnail/fobnail-platform-owner/issues/3

Reply:
@arturkow2000 agree. Plan have to be created and issue pointed by @krystian-hebel can be linked in documentation as point where we will try to create plan. Or maybe we need dedicated issues just for customizable policies with above issue mentioned as related.

Reply:
choose what works for you better